starter-code/pizza.js

```diff
@@ -17,69 +17,144 @@ var state = {
   greenPeppers: true,
   whiteSauce: false,
   glutenFreeCrust: false
-}
+};
 
 // This function takes care of rendering the pizza based on the state
 // This function is triggered once at the begining and everytime the state is changed
 function renderEverything() {
-  renderPepperonni()
-  renderMushrooms()
-  renderGreenPeppers()
-  renderWhiteSauce()
-  renderGlutenFreeCrust()
-
-  renderButtons()
-  renderPrice()
+  renderPepperonni();
+  renderMushrooms();
+  renderGreenPeppers();
+  renderWhiteSauce();
+  renderGlutenFreeCrust();
+
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll(".pep").forEach(function($pep) {
     if (state.pepperonni) {
       $pep.style.visibility = "visible";
-    }
-    else {
+    } else {
       $pep.style.visibility = "hidden";
     }
-  })
+  });
 }
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll(".mushroom").forEach(function($mush) {
+    if (state.mushrooms) {
+      $mush.style.visibility = "visible";
+    } else {
+      $mush.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(function($pepper) {
+    if (state.greenPeppers) {
+      $pepper.style.visibility = "visible";
+    } else {
+      $pepper.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll(".sauce").forEach(function($white) {
+    if (state.whiteSauce) {
+      $white.style.visibility = "visible";
+    } else {
+      $white.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGlutenFreeCrust() {
+  const glutenFree = document.querySelector(".crust");
+
+  if (state.glutenFreeCrust) {
+    glutenFree.classList.add("crust-gluten-free");
+  } else {
+    glutenFree.classList.remove("crust-gluten-free");
+  }
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
 }
 
+const ingredientClasses = {
```

Comment:
In the best case, you wouldn't want to rewrite code to make it fit for your function. Have a look at the method Object.keys() 

- if you would write 
[let stateArr = Object.keys(state)](url)
You will get an array with all the keys of the state object. If you need more help with this let me know, I am happy to help you!